Update README.md 및index.js 코드 초기화

antaehyeon opened the pull request:
- Update README.md
- index.js 코드 초기화

hannarita committed:
다각형의 넓이 구하기 완료함

hannarita committed:
다각형의 넓이를 계산하는 함수 완료함

hannarita committed:
다각형 넓이 구하기 함수의 인수 조건을 변경함

hannarita committed:
modify: declaration var size in function calculateCircleSize

hannarita committed:
modify: conditional statement for checking type in calculateRectSize

crongro committed:
Merge pull request #5 from hannarita/hannarita

Hannarita/다각형의 넓이 구하는 함수 만들기 완료

antaehyeon committed:
Update README.md 및index.js 코드 초기화

antaehyeon committed:
다각형의 넓이중 다각형의 넓이 코드 작성

antaehyeon committed:
다각형의 넓이 중 파라미터를 체크하는 다각형의 넓이함수 첫번째 코드 작성

antaehyeon committed:
다각형의 넓이 중 여러함수를 호출하는 함수 코드 작성

antaehyeon committed:
다각형의 넓이 중 배열을 이용한 로깅 코드 작성

antaehyeon committed:
Update README.md

crongro commented on index.js:
```diff
@@ -0,0 +1,123 @@
+function calCircleArea (radius) {
+    calledFunctionCount++;
+    calledFunctionSequence.push('Circle');
+    if (arguments.length === 0)
```
if/else를 사용할때 {}를 항상 사용하도록 해보세요.

antaehyeon replied:
확인하였습니다 👍

crongro commented on index.js:
```diff
@@ -0,0 +1,123 @@
+            calledFunctionCount++;
+            console.log("계산이 " + calledFunctionCount + "번 일어났습니다.");
+        }
+        let lastData = arguments[1];
```
이런 변수 선언은 보통 함수위쪽이나, 지금은 조건문 위쪽(else if 다음)에 선언해두는 게 일반적입니다.
중간에 선언하면 사용될 변수들이 잘 드러나지 않을 수 있어서요.

antaehyeon replied:
확인하였습니다 👍

crongro commented on index.js:
```diff
@@ -0,0 +1,123 @@
+
+// 사다리꼴 넓이 구하는 함수
+function calTrapezoidArea(lowerbase, upperbase, height) {
+    calledFunctionCount++;
```
값을 추가하고, 배열에 추가한 건 좋은데요.
생각해볼만 건, 이 부분이 함수가 잘 실행됐을때 처리해야 하는 것인지, 
아니만 함수의 정상처리를 떠나서 무조건 실행되는 것인지, 
를 결정하고 사용해야겠네요.

지금은 무조건 실행되는 코드니까, 의도한 게 맞다면 잘 하셨고요.

antaehyeon replied:
이 부분이 혹시 calledFunctionCount 가 증가하는 부분을 말씀하신 건가요?

crongro commented on index.js:
```diff
@@ -0,0 +1,123 @@
+    if (arguments[0] <= 0 || arguments[1] <= 0 || arguments[2] <= 0)
+        return "밑변 및 윗변, 높이의 길이는 0보다 커야합니다";
+
+    if (typeof(lowerbase) === "number" && typeof(upperbase) && typeof(height) === "number")
```
typeof(upperbase) 중간에 이건 이상한데요. 비교하는 게 없네요.

antaehyeon replied:
확인하였습니다 👍

crongro commented on index.js:
```diff
@@ -0,0 +1,123 @@
+}
+
+function getReport() {
+    calledFunctionSequence.forEach(value => console.log(value));
```
arrow 함수는 es6기능입니다. 잘 이해하고 사용하세요~
forEach도 마찬가지고요.

antaehyeon replied:
ES5 와 ES6의 차이에 대해서 명확하게 짚어가면서 코딩하도록 하겠습니다 👍

crongro commented on index.js:
```diff
@@ -0,0 +1,123 @@
+    if (arguments[0] <= 0 || arguments[1] <= 0)
+        return "가로 및 세로 길이는 0보다 커야합니다";
+
+    if (typeof(width) === "number" && typeof(height) === "number")
```
이렇게 동작하는 isAllNumber 함수를 만들어보실래요? 

if( isAllNumber([width, height]) ) 
if( isAllNumber([width]) )
if( isAllNumber([width, height, otherValue]) )

antaehyeon replied:
확인하였습니다. 새로 Push 하도록 하겠습니다 👍

crongro reviewed (changes requested):
인자 체크를 꼼꼼하게 하신 거 같네요.
좀더 수정해보시죠.

antaehyeon committed:
[Fix] 다각형의 넓이 코드에 대한 피드백을 반영한다

- if/else 를 사용할 때 괄호를 사용
- 변수들을 중간에 선언하지말고, 최상위나 조건문 위쪽에 선언
- 비교문 에러 수정
- ES5/ES6 구문확인
- isAllNumber 함수 생성

antaehyeon commented:
피드백을 반영한 코드를 Push 하였습니다 :)

crongro commented on index.js:
```diff
@@ -38,10 +41,34 @@ function calRectangleArea(width, height) {
     if (arguments[0] <= 0 || arguments[1] <= 0)
```
calCircleArea 함수에서, 오류 상황은 함수초반에 모아서, 빨리 반환처리하는 게 좋겠어요.

antaehyeon replied:
확인하였습니다 👍

crongro commented on index.js:
```diff
@@ -38,10 +41,34 @@ function calRectangleArea(width, height) {
+
+    for (var i = 0; i < arguments.length; i++) {
+        if (typeof (arguments[i]) === "number") {
+            correctNum++;
```
count를 세지 않고도 할 수 있을 거 같아요. 문제가 되면 바로 return!

antaehyeon replied:
코드를 생각나는대로만 작성해버렸네요.
앞으로 한번 더 확인하는 습관을 가지도록 하겠습니다.
확인하였습니다 👍

crongro reviewed (changes requested):
리뷰적었네요.

antaehyeon committed:
[Fix] 다각형의 넓이 코드에 대한 피드백을 반영한다

- 오류상황은 함수초반에 모아서 바로 return
- 코드 작성할 때 더 간단한 방법은 없는지 생각하면서 코딩하기

crongro reviewed (approved):
수고했어요.
